### PR TITLE
added basic implementation for GetDesiredState

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2362.

### CODE REVIEW

File: service/resource/vaultrolev1/current.go [hunk 1]
```diff
@@ -3,9 +3,10 @@ package vaultrolev1
 import (
 	"context"
 
-	"github.com/giantswarm/cert-operator/service/keyv2"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/vaultrole"
+
+	"github.com/giantswarm/cert-operator/service/keyv2"
```
Comment (xh3b4sd) [inline service/resource/vaultrolev1/current.go]: Boyscouting.

File: service/resource/vaultrolev1/desired.go [hunk 2]
```diff
@@ -2,8 +2,37 @@ package vaultrolev1
...
+	}
+
+	role := &vaultrole.Role{
+		AllowBareDomains: keyv2.AllowBareDomains(customObject),
+		AllowSubdomains:  AllowSubdomains,
+		AltNames:         keyv2.AltNames(customObject),
+		ID:               keyv2.ClusterID(customObject),
+		Organizations:    keyv2.Organizations(customObject),
+		TTL:              TTL,
+	}
```
Comment (xh3b4sd) [inline service/resource/vaultrolev1/desired.go]: That is basically it.

File: service/resource/vaultrolev1/desired_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	"testing"
+	"time"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/giantswarm/vaultrole"
+	"github.com/giantswarm/vaultrole/vaultroletest"
+)
+
+func Test_Resource_VaultRole_GetDesiredState(t *testing.T) {
```
Comment (xh3b4sd) [inline service/resource/vaultrolev1/desired_test.go]: And some fancy tests.

File: service/resource/vaultrolev1/resource.go [hunk 4]
```diff
@@ -8,7 +8,10 @@ import (
 )
 
 const (
-	Name = "vaultrolev1"
+	// AllowSubdomains defines whether to allow the generated root CA of the PKI
+	// backend to allow sub domains as common names.
+	AllowSubdomains = true
```
Comment (xh3b4sd) [inline service/resource/vaultrolev1/resource.go]: This seems to be the standard here. We do not configure it via CROs.

Comment (kopiczko) [reply service/resource/vaultrolev1/resource.go]: I don't see a reason to extract a constant. I'd just describe variable inside a function to not jump around.

Also I would make those constants private.

Comment (xh3b4sd) [reply service/resource/vaultrolev1/resource.go]: I always like this for godoc reasons. When you check the package you directly see that it is the hardcoded default.

**Pull request merged**